### PR TITLE
Migrate `expression_compiler_service_test.dart` to null-safety

### PR DESCRIPTION
**Pull request by elliette**

### CODE REVIEW

File: dwds/lib/src/services/expression_compiler_service.dart [hunk 1]
```diff
@@ -30,7 +30,7 @@ class _Compiler {
   Future<Map<String, Object>> _send(Map<String, Object> request) async {
     _sendPort.send(request);
     return await _responseQueue.hasNext
-        ? await _responseQueue.next as Map<String, Object>
+        ? Map<String, Object>.from(await _responseQueue.next)
```
Comment (annagrin) [inline dwds/lib/src/services/expression_compiler_service.dart:33]: Just to be safe (and not crash if the type is incorrect), can we check if the type of `await _responseQueue.next` is `Map<String, Object>` before creating a new map (or just returning the same object)? In the case where it is not the right type, we can just return a map with 'succeeded': 'false' and an error indicating that we got a bad response from the expression compiler worker.

**Review by annagrin — COMMENTED** [review]
Thanks Elliott! Left a comment asking for an additional check that probably should have been present there from the beginning:)

File: dwds/lib/src/services/expression_compiler_service.dart (hunk 1 above)
Comment (elliette) [reply dwds/lib/src/services/expression_compiler_service.dart]: Made the change but unfortunately the test is now failing, so we are returning the wrong type at some place. I'm trying to figure out where that is happening, do you have any ideas? The `next` value with the wrong type is of type `_InternalLinkedHashMap<String, dynamic>` with the value `{succeeded: true}`

Comment (elliette) [reply dwds/lib/src/services/expression_compiler_service.dart]: Ah I think it is here: https://github.com/dart-lang/sdk/blob/main/pkg/dev_compiler/lib/src/kernel/expression_compiler_worker.dart#L524. We should probably cast to the right type there?

Comment (annagrin) [reply dwds/lib/src/services/expression_compiler_service.dart]: Hmm it should be castable to Map but I think the problem is the difference in type parameters. Let's keep your version for now, I will try to add a check later.

**Comment by annagrin** [comment]
Thanks Elliott, sorry for the detour into type checking. Let's revert the last commit and keep your version.

**Comment by elliette** [comment]
> Thanks Elliott, sorry for the detour into type checking. Let's revert the last commit and keep your version.

Done!

**Review by annagrin — APPROVED** [review]
LGTM!

**Pull request merged**